include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -149,20 +139,30 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
 {
     template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3, typename _Compare>
     auto
-    operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3, _Compare __comp) const
+    operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3,
+               const _Compare& __comp) const
     {
-        using _Size = oneapi::dpl::__internal::__difference_t<_Range3>;
-
-        auto __n1 = __rng1.size();
-        auto __n2 = __rng2.size();
+        const _IdType __n1 = __rng1.size();
+        const _IdType __n2 = __rng2.size();
+        const _IdType __n = __n1 + __n2;
 
         assert(__n1 > 0 || __n2 > 0);
 
         _PRINT_INFO_IN_DEBUG_MODE(__exec);
 
-        const std::uint8_t __chunk = __exec.queue().get_device().is_cpu() ? 128 : 4;
-        const _Size __n = __n1 + __n2;
-        const _Size __steps = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __chunk);
+        // Empirical number of values to process per work-item
+        std::uint16_t __chunk = 128;
+        if (__exec.queue().get_device().is_gpu())
+        {
+            if (__n >= 16'777'216)
+                __chunk = 256;
+            else if (__n > 4'194'304)
+                __chunk = 8;
+            else
+                __chunk = 4;
+        }
```

Comment:
I've tested several values and I think there are two main areas:

1. We are cache bound for small arrays and can use a small chunk to occupy the GPU.
2. We are memory bound for large arrays and larger chunks seem to improve the global memory throughput.

I agree this might be overly specific to PVC. The middle step doesn't improve performance much either. I've reverted the change.